src/components/Home.jsx

```diff
@@ -5,29 +5,59 @@ import Upload from './Upload';
 
 const Home = ({ setActors, actors, currentScript, setCurrentScript }) => {
   const [titles, setTitles] = useState([]);
+  const [deleteError, setDeleteError] = useState('');
+  const [addError, setAddError] = useState('');
   const newActorInput = useRef('');
 
   // submit of actor name handler
   const newActorHandler = (event) => {
     event.preventDefault();
     const newActor = { name: newActorInput.current.value };
 
-    // Add the new actor to the database
+    // Add the new actor to the database, returning the new actor object
     fetch('/actors', {
       method: 'POST',
       headers: {
         'Content-Type': 'application/json',
       },
       body: JSON.stringify(newActor),
     })
-      .then(() => {
-        // set actors in state to be our new actors
-        setActors([...actors, newActor]);
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        } else {
+          throw new Error('Error when adding the actor');
```

Comment:
Would be a nice improvement to add context of some sort from the failed request so debugging is easier.